airflow-core/src/airflow/models/serialized_dag.py

```diff
@@ -648,11 +650,18 @@ def load_json(deps_data):
             .join(cls.dag_model)
             .where(~DagModel.is_stale)
         )
-        iterator = (
-            [(dag_id, load_json(deps_data)) for dag_id, deps_data in query]
-            if load_json is not None
-            else query.all()
-        )
+
+        if load_json is not None:
+            iterator: Sequence[tuple[str, dict[Any, Any]]] = [
+                (dag_id, load_json(deps_data)) for dag_id, deps_data in query
+            ]
+        else:
+            raw_results = query.all()
+            iterator = [
+                (row[0], json.loads(row[1]) if isinstance(row[1], str) else row[1] if row[1] else {})
```

Comment:
would be nice if we could add a function or some comment here. it's not that easy to read. Thanks!